Content.Server/GameObjects/Components/Disposal/DisposalUnitComponent.cs

```diff
@@ -136,10 +138,20 @@ public bool CanInsert(IEntity entity)
                 return false;
             }
 
-            if (!entity.TryGetComponent(out IPhysicsComponent? physics) ||
+            IDamageableComponent? damageState;
+            entity.TryGetComponent<IDamageableComponent>(out damageState);
+
+            if (!entity.TryGetComponent(out IPhysicsComponent? physics) || 
                 !physics.CanCollide)
             {
-                return false;
+                if (damageState != null) {
+                    if (damageState.CurrentState != DamageState.Dead) {
```

Comment:
Merge the TryGetComponent with the `damageState` declaration, and also put the `TryGetComponent` where the current null check is. Afterwards, you can put the last if after the `TryGetComponent` using an and.